third-party/realdds/src/dds-device-impl.cpp

```diff
@@ -129,6 +129,11 @@ void dds_device::impl::open( const dds_stream_profiles & profiles )
         stream_profiles[stream->name()] = profile->to_json();
 
         _streams[stream->name()]->open( _info.topic_root + '/' + stream->name(), _subscriber );
+        if( _md_supported )
+        {
+            std::string metadata_stream_name = stream->name() + " metadata";
+            _streams[metadata_stream_name]->open( _info.topic_root + '/' + metadata_stream_name, _subscriber );
```

Comment:
So, if the device supports metadata, ALL the streams need to have a metadata stream?

Reply:
Currently yes.
If device supports metadata all streams can have metadata attached to them. Especially as we send information like timetag and frame_id to streams that don't include this information in the data (e.g., IMU)

Reply:
I think this is too much. Why not make the metadata stream-based and be more flexible?

Reply:
I think we really need to discuss this... I'm not sure we should just continue as-is without making a decision

Reply:
Implemented per-stream metadata support.